core/net/common.odin

```diff
@@ -82,10 +90,16 @@ Network_Error :: union {
 	Socket_Option_Error,
 	Parse_Endpoint_Error,
 	Resolve_Error,
+	DNS_Error,
+}
+
+
+Resolve_Error :: enum {
+	Unable_To_Resolve = 1,
 }
 
 DNS_Error :: enum {
-	Invalid_Hostname_Error,
+	Invalid_Hostname_Error = 1,
```

Comment:
Ideally, I'd add an `#assert` to `Network_Error` to ensure none of the enums have a named zero value, but I don't know quite how I'd approach that.
Maybe we want `intrinsics.type_union_variants_have_value(Network_Error, nil)` 😄 

I could add such an assert to every enum individually, but that kinda gross honestly...